cvmfs/sync_mediator.cc

```diff
@@ -985,10 +985,9 @@ void SyncMediator::AddLocalHardlinkGroups(const HardlinkGroupMap &hardlinks) {
        iEnd = hardlinks.end(); i != iEnd; ++i)
   {
     if (i->second.hardlinks.size() != i->second.master->GetUnionLinkcount()) {
-      LogCvmfs(kLogPublish, kLogStdout, "Hardlinks across directories (%s)",
+      LogCvmfs(kLogPublish, kLogDebug, "Hardlinks across directories (%s)",
                i->second.master->GetUnionPath().c_str());
-      if (!params_->ignore_xdir_hardlinks)
-        PANIC(NULL);
+      if (!params_->ignore_xdir_hardlinks) PANIC(NULL);
```

Comment:
```suggestion
      if (!params_->ignore_xdir_hardlinks) {
        PANIC(kLogSyslogErr | kLogDebug, "Hardlinks across directories (%s)",
              i->second.master->GetUnionPath().c_str());
       }
```

(And then we can actually get rid of the previous debug log statement.  My bad for not suggesting this right away.

Reply:
Should we merge the two individual if statements `if (i->second.hardlinks.size() != i->second.master->GetUnionLinkcount())` and `if (!params_->ignore_xdir_hardlinks)` into a single one with `&&`? @jblomer

Reply:
I think both ways are fine, it's up to you